tools/src/command/host.rs

```diff
@@ -0,0 +1,166 @@
+        fp1,
+        concat!(
+            "\n",
+            "[workspace]\n",
+            "members = [\n",
+            "    \"src/guest\"\n",
+            "]\n\n",
+            "# Generated by nexus-tools, do not remove!\n",
+            "#\n",
+            "# This profile is used for generating proofs, as Nexus VM support for compiler optimizations is still under development.\n",
+            "[profile.release-unoptimized]\n",
+            "inherits = \"release\"\n\n",
+            "[profile.release-unoptimized.package.guest]\n",
+            "opt-level = 0\n\n",
+
+            // https://doc.rust-lang.org/1.58.1/cargo/reference/overriding-dependencies.html#working-with-an-unpublished-minor-version
+            "# These patches are required for some of the underlying cryptography libraries used by Nexus.\n",
+            "[patch.crates-io]\n",
+            "ark-crypto-primitives = {{ git = \"https://github.com/arkworks-rs/crypto-primitives/\", rev = \"d27a5c8\" }}\n",
+            "ark-r1cs-std = {{ git = \"https://github.com/arkworks-rs/r1cs-std/\", rev = \"2ca3bd7\" }}\n",
+            "ark-ff = {{ git = \"https://github.com/arkworks-rs/algebra/\", rev = \"2a80c54\" }}\n",
+            "ark-ec = {{ git = \"https://github.com/arkworks-rs/algebra/\", rev = \"2a80c54\" }}\n",
+            "ark-serialize = {{ git = \"https://github.com/arkworks-rs/algebra/\", rev = \"2a80c54\" }}\n",
+            "ark-poly = {{ git = \"https://github.com/arkworks-rs/algebra/\", rev = \"2a80c54\" }}\n",
+            "ark-test-curves = {{ git = \"https://github.com/arkworks-rs/algebra/\", rev = \"2a80c54\" }}\n",
+            "ark-poly-commit = {{ git = \"https://github.com/arkworks-rs/poly-commit/\", rev = \"12f5529\" }}\n",
+            "ark-bn254 = {{ git = \"https://github.com/arkworks-rs/curves/\", rev = \"8c0256a\" }}\n",
+            "ark-pallas = {{ git = \"https://github.com/arkworks-rs/curves/\", rev = \"8c0256a\" }}\n",
+            "ark-vesta = {{ git = \"https://github.com/arkworks-rs/curves/\", rev = \"8c0256a\" }}\n",
+            "ark-bls12-381 = {{ git = \"https://github.com/arkworks-rs/curves/\", rev = \"3fded1f\" }}"
+        )
+    )?;
+
+    // src/main.rs
+    fs::write(
+        host_path.join("src/main.rs"),
+        HOST_TEMPLATE_SRC_MAIN
+            .replace(
+                "const PACKAGE: &str = \"example\"",
+                "const PACKAGE: &str = \"guest\"",
+            )
+            .replace(
+                "const EXAMPLE: &str = \"example\"",
+                "const EXAMPLE: &str = \"guest\"",
+            ),
+    )?;
+
+    cargo(None, ["new", guest_str])?;
+    cargo(
+        Some(&guest_path),
+        [
+            "add",
+            "--git",
+            "https://github.com/nexus-xyz/nexus-zkvm.git",
+            "nexus-rt",
+        ],
+    )?;
+
+    // add postcard because it is used for (de)serializing from/to the input/output tapes
+    cargo(Some(&guest_path), ["add", "postcard", "-F", "alloc"])?;
```

Comment:
I think it's better to re-export from sdk or wrap it into public functions.

Overall I think the host functionality is not going to be easy to maintain, but here's the best effort.